### ISSUE TITLE
Add PEP517 support

### ISSUE THREAD

**Issue by AstraLuma** [devpi/devpi#834]
`devpi upload` does not currently support non-setuptools pep517 projects using pyproject.toml (eg, those built with poetry):

```
$ devpi upload
no setup.py found in /home/astraluma/src/project
```

The work around is to do this in two steps:

```
$ poetry build  # or use build, bork, or another pep517 wrapper
$ devpi upload dist/*
```

**Comment by fschulze** [comment]
Could you link to a public project which uses this, which I can use for testing?

**Comment by AstraLuma** [comment]
Any project that uses poetry.

Some of mine:
* https://github.com/buildahscript/buildahscript-py
* https://github.com/ppb/binny
* https://github.com/astraluma/clomp
* https://github.com/minecraft-podman/podcraft
* https://github.com/minecraft-podman/manage

**Comment by link2xt** [comment]
At least the configuration section `[devpi:upload]` should be moved from `setup.cfg` to the corresponding `pyproject.toml` tool section.

**Comment by hungrybirder** [comment]
Look forward to this feature.

**Comment by fschulze** [comment]
See
https://m.devpi.net/fschulze/dev/devpi-client/6.0.0.dev9
and
https://m.devpi.net/fschulze/dev/+f/5fe/c5286f35d0d8a/devpi_client-6.0.0.dev9-py2.py3-none-any.whl

**Comment by fschulze** [comment]
Released with devpi-client 6.0.0.
